File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/validator/ValidationResult.java
package cz.fb.manaus.reactor.betting.validator;

public enum ValidationResult {
    ACCEPT {
        @Override
        public boolean isSuccess() {
            return true;
        }
    }, REJECT {
        @Override
        public boolean isSuccess() {
            return false;
        }
    };

    public static ValidationResult of(boolean condition) {
        if (condition) {
            return ACCEPT;
        } else {
            return REJECT;
        }
    }

    public abstract boolean isSuccess();

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/validator/ValidationResult.kt
package cz.fb.manaus.reactor.betting.validator

enum class ValidationResult(val isSuccess: Boolean) {
    ACCEPT(true),
    REJECT(false),
    SKIP(false);

    companion object {
        fun of(condition: Boolean): ValidationResult {
            return if (condition) {
                ACCEPT
            } else {
                REJECT
            }
        }
    }

}
